src/view/reviewManager.ts

```diff
@@ -76,8 +82,30 @@ export class ReviewManager implements vscode.DecorationProvider {
 			vscode.commands.executeCommand('vscode.open', vscode.Uri.file(nodePath.resolve(this._repository.path, params.path)), opts);
 		}));
 		this._disposables.push(_repository.onDidRunGitStatus(e => {
-			// todo, validate state only when state changes.
-			this.updateState();
+			const oldHead = this._previousRepositoryState.HEAD;
+			const newHead = this._repository.HEAD;
+			const sameUpstream = !!oldHead.upstream
+				? newHead.upstream && oldHead.upstream.name === newHead.upstream.name && oldHead.upstream.remote === newHead.upstream.remote
+				: !!newHead.upstream;
+			const sameHead = oldHead.ahead === newHead.ahead
+				&& oldHead.behind === newHead.behind
+				&& oldHead.commit === newHead.commit
+				&& oldHead.name === newHead.name
+				&& oldHead.remote === newHead.remote
+				&& oldHead.type === newHead.type
+				&& sameUpstream;
+
+			const sameRemotes = this._previousRepositoryState.remotes.length === this._repository.remotes.length
```

Comment:
We may want to filter out remotes created for PR purpose first so a fresh checkout will only lead to one `updateState`, triggered by branch change.

Reply:
I'm going to hold off on making that change now, but I agree